src/main/scala/viper/gobra/frontend/info/implementation/typing/TypeTyping.scala

```diff
@@ -179,22 +177,25 @@ trait TypeTyping extends BaseTyping { this: TypeInfoImpl =>
       }
   }
 
-  private def structSymbType(t: PStructType): Type = {
-    def infoFromFieldDecl(f: PFieldDecl, isGhost: Boolean): StructFieldT = StructFieldT(typeSymbType(f.typ), isGhost)
+  private[typing] def structSymbType(t: PStructType, isGhost: Boolean): StructT = {
+    def infoFromFieldDecl(f: PFieldDecl, isFieldGhost: Boolean): StructFieldT = StructFieldT(typeSymbType(f.typ), isFieldGhost)
 
-    def makeFields(x: PFieldDecls, isGhost: Boolean): ListMap[String, StructClauseT] = {
-      x.fields.foldLeft(ListMap[String, StructClauseT]()) { case (prev, f) => prev + (f.id.name -> infoFromFieldDecl(f, isGhost)) }
+    def makeFields(x: PFieldDecls, areFieldsGhost: Boolean): ListMap[String, StructClauseT] = {
+      x.fields.foldLeft(ListMap[String, StructClauseT]()) { case (prev, f) => prev + (f.id.name -> infoFromFieldDecl(f, areFieldsGhost)) }
     }
-    def makeEmbedded(x: PEmbeddedDecl, isGhost: Boolean): ListMap[String, StructClauseT] =
-      ListMap(x.id.name -> StructEmbeddedT(miscType(x.typ), isGhost))
 
+    def makeEmbedded(x: PEmbeddedDecl, isEmbeddedGhost: Boolean): ListMap[String, StructClauseT] =
+      ListMap(x.id.name -> StructEmbeddedT(miscType(x.typ), isEmbeddedGhost))
```

Comment:
ghost embeddings sound dangerous, as they might modify the method set of a type. In particular, I wonder if this program would be accepted by gobra:
```go
type T1 struct {
    // ...
}

func (t T1) F() {
    //...
}

type T struct {
    ghost T1
    // ...
}

func test(t T) {
    t.F() // this should be rejected by gobra, because the fact that T has a method F() comes from a ghost embedding, and we are in actual code
}
```

Reply:
(this example is accepted in Gobra currently)